### ISSUE TITLE
How to upgrade zammad from 5.3.1-6 to 6.2

### ISSUE THREAD

**Issue by levanluu** [zammad/zammad#5040]
### Used Zammad Version

5.3.1-6

### Actual behaviour

I want to upgrade my current version of zammad to the latest version, but I've always had problems with the zammad-init step.

Is there any way I can upgrade it?

I checked the upgrade to each zammad version and it failed from Zammad version 5.4.0

### Expected behaviour

`Caused by: ActiveRecord::RecordInvalid: Validation failed: User must exist /usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/validations.rb:80:in raise_validation_error'
/usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/validations.rb:53:in save!' /usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/transactions.rb:302:in block in save!'
/usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/transactions.rb:354:in block in with_transaction_returning_status' /usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/connection_adapters/abstract/database_statements.rb:314:in transaction'
/usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/transactions.rb:350:in with_transaction_returning_status' /usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/transactions.rb:302:in save!'
/usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/suppressor.rb:54:in save!' /opt/zammad/db/migrate/20230129220648_taskbar_update_preference_tasks.rb:11:in block in up'
/usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/relation/batches/batch_enumerator.rb:54:in each' /usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/relation/batches/batch_enumerator.rb:54:in block in each_record'
/usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/relation/batches.rb:245:in block in in_batches' /usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/relation/batches.rb:229:in loop'
/usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/relation/batches.rb:229:in in_batches' /usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/relation/batches/batch_enumerator.rb:53:in each'
/usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/relation/batches/batch_enumerator.rb:53:in each_record' /opt/zammad/db/migrate/20230129220648_taskbar_update_preference_tasks.rb:8:in up'
/usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/migration.rb:908:in public_send' /usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/migration.rb:908:in exec_migration'
/usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/migration.rb:889:in block (2 levels) in migrate' /usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/migration.rb:888:in block in migrate'
/usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/connection_adapters/abstract/connection_pool.rb:215:in with_connection' /usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/migration.rb:887:in migrate'
/usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/migration.rb:1095:in migrate' /usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/migration.rb:1409:in block in execute_migration_in_transaction'
/usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/connection_adapters/abstract/transaction.rb:319:in block in within_new_transaction' /usr/local/bundle/gems/activesupport-7.0.8/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in handle_interrupt'
/usr/local/bundle/gems/activesupport-7.0.8/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in block in synchronize' /usr/local/bundle/gems/activesupport-7.0.8/lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in handle_interrupt'
/usr/local/bundle/gems/activesupport-7.0.8/lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in synchronize' /usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/connection_adapters/abstract/transaction.rb:317:in within_new_transaction'
/usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/connection_adapters/abstract/database_statements.rb:316:in transaction' /usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/transactions.rb:209:in transaction'
/usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/migration.rb:1460:in ddl_transaction' /usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/migration.rb:1408:in execute_migration_in_transaction'
/usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/migration.rb:1382:in each' /usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/migration.rb:1382:in migrate_without_lock'
/usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/migration.rb:1331:in block in migrate' /usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/migration.rb:1481:in block in with_advisory_lock'
/usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/connection_adapters/abstract/connection_pool.rb:215:in with_connection' /usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/migration.rb:1496:in with_advisory_lock_connection'
/usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/migration.rb:1477:in with_advisory_lock' /usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/migration.rb:1331:in migrate'
/usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/migration.rb:1166:in up' /usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/migration.rb:1141:in migrate'
/usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/tasks/database_tasks.rb:262:in migrate' /usr/local/bundle/gems/activerecord-7.0.8/lib/active_record/railties/databases.rake:92:in block (2 levels) in

'
/usr/local/bundle/gems/bundler-2.4.1/lib/bundler/cli/exec.rb:58:in load' /usr/local/bundle/gems/bundler-2.4.1/lib/bundler/cli/exec.rb:58:in kernel_load'
/usr/local/bundle/gems/bundler-2.4.1/lib/bundler/cli/exec.rb:23:in run' /usr/local/bundle/gems/bundler-2.4.1/lib/bundler/cli.rb:491:in exec'
/usr/local/bundle/gems/bundler-2.4.1/lib/bundler/vendor/thor/lib/thor/command.rb:27:in run' /usr/local/bundle/gems/bundler-2.4.1/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in invoke_command'
/usr/local/bundle/gems/bundler-2.4.1/lib/bundler/vendor/thor/lib/thor.rb:392:in dispatch' /usr/local/bundle/gems/bundler-2.4.1/lib/bundler/cli.rb:34:in dispatch'
/usr/local/bundle/gems/bundler-2.4.1/lib/bundler/vendor/thor/lib/thor/base.rb:485:in start' /usr/local/bundle/gems/bundler-2.4.1/lib/bundler/cli.rb:28:in start'
/usr/local/bundle/gems/bundler-2.4.1/exe/bundle:45:in block in <top (required)>' /usr/local/bundle/gems/bundler-2.4.1/lib/bundler/friendly_errors.rb:117:in with_friendly_errors'
/usr/local/bundle/gems/bundler-2.4.1/exe/bundle:33:in <top (required)>' /usr/local/bundle/bin/bundle:25:in load'
/usr/local/bundle/bin/bundle:25:in <main>' Tasks: TOP => db:migrate`

**Comment by WWoodpecker** [comment]
Hello, I did it as follows:
- Moved all files to the database in Zammad (https://github.com/zammad/zammad/issues/428)
- Created a backup
- Moved Docker volumes away (2nd backup ^^)
- SnapShot VM (paranoia backup :D)

Here is my KB entry on my knowledge base ^^
- Copy and unpack SQL backup data from Zammad backup container to Postgres container Data
- Backup data (SQL and file) are located in /data/zammad/backup
- Start the Postgres container and connect to the bash
-- docker exec -it zammad-postgresql bash
- Delete the old database (note, this is irrevocable) and create a new one
-- dropdb zammad_production -U zammad
-- createdb zammad_production -U zammad
- Restore old database backup
-- psql zammad_production < /var/lib/postgresql/data/backup.psql --username zammad --dbname zammad_production
- Restore file backup to /data/zammad/storage (should actually be nothing!)
- Start all containers of the Zammad instance

**Comment by levanluu** [comment]
I did all the steps above, and when I started all containers, the init container could not execute the migration command

**Comment by levanluu** [comment]
https://github.com/zammad/zammad-docker-compose/releases/tag/v6.0.0
Steps I took to upgrade PostgreSQL from 9.6 to 15.0.0

Our problem is that we cannot run migrations to update tables or add columns

**Comment by dominikklein** [comment]
At least the error in the first article seems to be some problem with existing taskbar entries in your system.
I will try to shortly take a look on Monday.

**Comment by levanluu** [comment]
Hi @dominikklein , Do you have any findings about the issue I mentioned

**Comment by levanluu** [comment]
Hi @dominikklein, Can I run taskbar destroy_all and run Upgrade again, Will this break my system if, in bad case I run Migration without success

**Comment by levanluu** [comment]
My issue is similar to this ticket, but this ticket is still unresolved
https://github.com/zammad/zammad/issues/4669

**Comment by levanluu** [comment]
Can anyone help me with this, I need your help

**Comment by lookslik1** [comment]
Hi @levanluu did you find a solution? I'm facing the same issue

**Comment by levanluu** [comment]
Sorry @lookslik1, It's so painful, I'm stuck here for a long time, waiting for help, my system has more than 200GB of data and I can't update the data structure because it will kill the whole system.

**Comment by MrGeneration** [comment]
We do not provide support / help in our repositories. Generally we only support the application itself.
I will give one hint though:
You have to update via Mayors. That means: 5.4 -> 6.0 -> 6.2.
Not less, not more.

You will also have to fetch the docker-compose differences (that is your responsibility). A lot has changed in between 5.4 and 6.2 docker wise.

**Comment by levanluu** [comment]
Yes i understand that. I just think you understand your system and I'm looking forward to your support. If you find this offensive, I will close this ticket

I updated our system from version 5.2 to 5.3 to 5.4 and it's stuck at version 5.4. I have already made the issue I mentioned before clear. Due to most of the problems I encountered, you suggested updating to version 6.2 , but I encountered problems updating to that version.
